website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+
+After a good time of work, we can finally see what we want on the screen, but we need a lot to create a good program!
+
+Another very important thing in graphics programming is the ability to draw images. In this tutorial, you'll learn:
+
+* What is a texture.
+* How to send more data to shaders.
+* How to use texture coordinates.
+* How to import images as textures.
+* How shaders handle images.
+* How to use texture parameters.
+* How to allow mipmaps.
+
+Like before, this tutorial will feature a lot of new content and info, so we'll take it slow and explain.
+
+## What is a texture
+Technically, textures are multidimensional objects that store color data.
+
+Mainly, textures store 2D data of the pixels of an image like a 2D array (this is just for comparison. Texture objects are a lot more
+complex than just 2D arrays!).
+This data can be used to be sanded to a shader, in an easy way for it can read.
+
+<?# Info "There's 1D or 3D texture objects, but we will focus on 2D" /?>
+
+But don't think texturing objects is as easy as a simple drag-and-drop! Textures need some information to be drawn.
+The most important information that we need to send to the texture to see it on the screen is the texture coordinates.
+The texture coordinates are the positions of the texture at a particular vertex. Below we will learn how to fill out
+this data and read it from a shader.
+
+
+## Sending more data to shaders
+With our new objective, let's learn how to send more data to shaders, and use it as texture coordinates.
+
+texture coordinates are a type of data that can be (and should be) manipulated by the vertex shader before be used by the fragment shader,
+so this is the way that we use to send the coordinates.
+
+Let's go back to our `OnLoad` function and update our `vertices` array to the following:
+```c#
+// The quad vertices data. Now with Texture coordinates!
+float[] vertices =
+{
+//       aPosition     | aTexCoords
+     0.5f,  0.5f, 0.0f,  1.0f, 1.0f,
+     0.5f, -0.5f, 0.0f,  1.0f, 0.0f,
+    -0.5f, -0.5f, 0.0f,  0.0f, 0.0f,
+    -0.5f,  0.5f, 0.0f,  0.0f, 1.0f
+};
+```
+
+With how we have structured the data, both the vertex and texture coordinates can be stored in a single array.
+
+If we try to run our program now...
+
+![A really messed up quad!](../../../images/opengl/chapter1/leasson3/messed-up-quad.png)
+
+"Oh no! We messed up with everything!" is what you should be thinking. And in reality, yes, we did. But it's simple to resolve, we just
+forgot to update our vertex layout definition.
+
+In our `VertexAttribPointer` calls, we declared a size of 3 and a stride of 3 for the `aPosition` attribute.
+Because of this, our buffer is being read like this:
+
+![Buffer reading wrong data](../../../images/opengl/chapter1/leasson3/wrong-pointers.png)
```

Comment:
```suggestion
![Buffer reading wrong data](../../../images/opengl/chapter1/leasson3/wrong-pointers.png)

As you can see, the vertex buffer is being read as if the first vertex was composed of the first three floats in the buffer, then the second vertex the next three floats, the third vertex the next three, and so on and so forth. As we can see however, this is now wrong! Since each vertex has two UV floats after its three position floats, this means that the first vertex's U and V floats are being read as the second vertex X and Y position floats!
```

I like this image a lot, well done!